openml-h2o/src/main/java/com/feedzai/openml/h2o/server/H2OApp.java

```diff
@@ -176,8 +195,38 @@ public Path export(final Model model,
         return exportDir;
     }
 
+    private Model trainAnomalyDetection(final MLAlgorithmDescriptor algorithmDescriptor,
+                                        final Frame trainingFrame,
+                                        final Map<String, String> params,
+                                        final long randomSeed, final DatasetSchema datasetSchema) throws ModelTrainingException {
+        final Job<M> modelJob;
+
+        final H2OAlgorithm h2OAlgorithm = getH2OAlgorithm(algorithmDescriptor);
+
+        switch (h2OAlgorithm) {
+            case ISOLATION_FOREST:
+                final IsolationForestV3.IsolationForestParametersV3 isolationForestParams = new H2OIsolationForestUtils().parseParams(trainingFrame, params, randomSeed, datasetSchema);
+
+                modelJob = train(() -> new IsolationForest(isolationForestParams.createAndFillImpl()).trainModel())
+                        .orElseThrow(() -> createModelTrainingException(algorithmDescriptor));
+                break;
+
+            default:
+                logger.error("Training not supported for algorithm {}", algorithmDescriptor.getAlgorithmName());
+                throw new IllegalArgumentException("Unsupported anomaly detection algorithm: " + algorithmDescriptor);
```

Comment:
What I normally do is have a String msg = String.format.... that I then reuse in the logger and exception. That way I avoid that in the future both go out of sync (e.g., right now, the log shows alg name and the exception already shows the full descriptor, so they are already out of sync.